Automattic/jetpack
# VideoPress: Add nudge to Video block

Fixes #14733

This will add an upgrade banner nudge to the core video block when using Jetpack on WordPress.com with a free plan.

![video-upgrade-1](https://user-images.githubusercontent.com/1464705/74962196-4be1b500-53c4-11ea-8608-780dc69bdc11.png)

#### Changes proposed in this Pull Request:
* When using a free plan on a WordPress.com site you will see the above banner when adding the core video block. The buttons will also be disabled. This makes it clear that you cannot upload a video file until you upgrade to a Premium plan.


#### Testing instructions:

* Apply D39101-code to your sandbox
* Insert the video block into a post or page on a WordPress.com site that does not have a paid plan (and thus no VideoPress access)
* Confirm that you see an upgrade banner (wording will be specific to WordPress.com)
* Confirm that the upgrade banner upgrade button takes you to the payment flow for a premium plan
* Confirm that you are not able to upload a video via the dropzone.
* Confirm that the "Upload File" and "Media Library" buttons are disabled.
* Confirm you can still use the URL option and embed a video via an external URL
* Confirm other paid blocks still show the correct upgrade banner with their original wording.
* Confirm that when on a WordPress.com site with a paid plan, the original video block placeholder is restored, there is no upgrade banner, and you can upload and embed videos with VideoPress.

#### Proposed changelog entry for your changes:
* No changes needed.

**TODO:**

- [x] Determine the correct plan check, test this on WordPress.com
- [x] Get upgrade button visible and linking to correct place on WordPress.com
- [x] Disable upload buttons when banner is present
- [x] Update the nudge component to accept a custom main title
- [x] Code review to determine the correct place to load this filter
